The DEPLOY_DIR_RPM might be on a read-only partition, so move the lock
to the TMPDIR.  This will allow a read-write location for that file.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

diff -ur oe-core.orig/meta/classes/package_rpm.bbclass oe-core/meta/classes/package_rpm.bbclass
--- oe-core.orig/meta/classes/package_rpm.bbclass	2013-01-22 16:11:11.000000000 -0600
+++ oe-core/meta/classes/package_rpm.bbclass	2013-01-22 17:11:21.900166304 -0600
@@ -1122,7 +1122,7 @@
 do_package_write_rpm[sstate-outputdirs] = "${DEPLOY_DIR_RPM}"
 # Take a shared lock, we can write multiple packages at the same time...
 # but we need to stop the rootfs/solver from running while we do...
-do_package_write_rpm[sstate-lockfile-shared] += "${DEPLOY_DIR_RPM}/rpm.lock"
+do_package_write_rpm[sstate-lockfile-shared] += "${TMPDIR}/rpm.lock"
 
 python do_package_write_rpm_setscene () {
     sstate_setscene(d)
diff -ur oe-core.orig/meta/classes/populate_sdk_rpm.bbclass oe-core/meta/classes/populate_sdk_rpm.bbclass
--- oe-core.orig/meta/classes/populate_sdk_rpm.bbclass	2013-01-22 16:11:10.000000000 -0600
+++ oe-core/meta/classes/populate_sdk_rpm.bbclass	2013-01-22 17:11:31.993978913 -0600
@@ -17,7 +17,7 @@
 RPMOPTS="--dbpath ${rpmlibdir}"
 RPM="rpm ${RPMOPTS}"
 
-do_populate_sdk[lockfiles] += "${DEPLOY_DIR_RPM}/rpm.lock"
+do_populate_sdk[lockfiles] += "${TMPDIR}/rpm.lock"
 
 populate_sdk_post_rpm () {
 
diff -ur oe-core.orig/meta/classes/rootfs_rpm.bbclass oe-core/meta/classes/rootfs_rpm.bbclass
--- oe-core.orig/meta/classes/rootfs_rpm.bbclass	2013-01-22 16:11:11.000000000 -0600
+++ oe-core/meta/classes/rootfs_rpm.bbclass	2013-01-22 17:11:47.438164221 -0600
@@ -41,7 +41,7 @@
 
 # RPM doesn't work with multiple rootfs generation at once due to collisions in the use of files 
 # in ${DEPLOY_DIR_RPM}. This can be removed if package_update_index_rpm can be called concurrently
-do_rootfs[lockfiles] += "${DEPLOY_DIR_RPM}/rpm.lock"
+do_rootfs[lockfiles] += "${TMPDIR}/rpm.lock"
 
 fakeroot rootfs_rpm_do_rootfs () {
 	set +x
diff -ur oe-core.orig/meta/classes/image.bbclass oe-core/meta/classes/image.bbclass
--- oe-core.orig/meta/classes/image.bbclass	2013-01-22 17:44:28.646166494 -0600
+++ oe-core/meta/classes/image.bbclass	2013-01-22 17:44:52.298166823 -0600
@@ -171,7 +171,7 @@
 
 do_rootfs[nostamp] = "1"
 do_rootfs[dirs] = "${TOPDIR}"
-do_rootfs[lockfiles] += "${IMAGE_ROOTFS}.lock"
+do_rootfs[lockfiles] += "${IMAGE_ROOTFS}/.lock"
 do_rootfs[cleandirs] += "${S}"
 do_build[nostamp] = "1"
 
